app/src/main/java/org/openedx/app/AppActivity.kt

```diff
@@ -136,6 +136,10 @@ class AppActivity : AppCompatActivity(), InsetHolder, WindowSizeHolder {
             }
         }
 
+        if (viewModel.canResetAppDirectory){
+            viewModel.resetAppDirectory(this)
+        }
+
```

Comment:
What if we just encapsulate all this inside of the `AppViewModel#onCreate`?
In this case, we would remove the extra variable and this logic from the view.

Reply:
sure, I'll update it accordingly.